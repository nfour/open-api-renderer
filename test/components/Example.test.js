import React from 'react'
import Example from './../../src/components/Example/Example'
import ReactShallowRenderer from 'react-test-renderer/shallow'

describe('<Example />', () => {
  const shallow = new ReactShallowRenderer()
  it('renders nothing when there is no examples passed', () => {
    const tree = shallow.render(
      <Example examples={[]} />
    )

    expect(tree).toBeNull()
  })

  it('renders a string example', () => {
    let tree = shallow.render(
      <Example examples={['<message>hi</message>']} />
    )

    expect(tree).toMatchSnapshot()
  })

  it('renders a JSON example', () => {
    const tree = shallow.render(
      <Example examples={[{'message': 'hi'}]} />
    )

    expect(tree).toMatchSnapshot()
  })
})
