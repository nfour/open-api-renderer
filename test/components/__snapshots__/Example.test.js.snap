// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Example /> renders a JSON example 1`] = `
<div
  className={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className=""
  >
    <span
      onClick={[Function]}
      onMouseOver={[Function]}
      title="Copy to Clipboard"
    >
      Copy
    </span>
  </div>
  <
    collapsed={false}
    displayDataTypes={false}
    displayObjectSize={false}
    enableClipboard={false}
    src={
      Object {
        "message": "hi",
      }
    }
    theme="chalk"
  />
</div>
`;

exports[`<Example /> renders a string example 1`] = `
<div
  className={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className=""
  >
    <span
      onClick={[Function]}
      onMouseOver={[Function]}
      title="Copy to Clipboard"
    >
      Copy
    </span>
    <span
      onClick={[Function]}
    >
      Expand All
    </span>
    <span
      onClick={[Function]}
    >
      Collapse All
    </span>
  </div>
  &lt;message&gt;hi&lt;/message&gt;
</div>
`;

exports[`<Example /> renders nothing when there is no examples passed 1`] = `null`;
